libs/shared/providers/src/txButton/components/TxButton.tsx

```diff
@@ -173,9 +183,11 @@ export const TxButton = <
           waitTxStatus === 'pending'
         ? waitingTxLabel ??
           intl.formatMessage({ defaultMessage: 'Processing Transaction' })
-        : label ?? capitalize(params.functionName);
+        : customButtonLabel ?? capitalize(params?.functionName ?? '');
   const isDisabled =
     disabled ||
+    !simulateData ||
+    !!simulateError ||
```

Comment:
that will prevent the user from clicking the button and dispatch the `onSimulateError`

Reply:
agg, was thinking that just happened automatically -- because we already know if there was a simulate error before the button is even clicked